pkg/oci/load_balancer.go

```diff
@@ -436,11 +436,17 @@ func (cp *CloudProvider) updateBackendSet(ctx context.Context, lbOCID string, ac
 
 		workRequestID, err = cp.client.LoadBalancer().CreateBackendSet(ctx, lbOCID, action.Name(), bs)
 	case Update:
-		err = cp.securityListManager.Update(ctx, lbSubnets, nodeSubnets, sourceCIDRs, listenerPort, backendPort, healthCheckPort)
-		if err != nil {
+		// FIXME(apryde): This is inelegant and inefficient. Update() should be refactored
+		// to take the old backend port and handle removal of associated rules.
+		if action.OldBackendSet != nil && *action.OldBackendSet.Backends[0].Port != backendPort {
+			oldBackendPort := *action.OldBackendSet.Backends[0].Port
+			if err = cp.securityListManager.Delete(ctx, lbSubnets, nodeSubnets, listenerPort, oldBackendPort, healthCheckPort); err != nil {
+				return errors.Wrapf(err, "deleteing security rule for old node port %d", oldBackendPort)
+			}
+		}
+		if err = cp.securityListManager.Update(ctx, lbSubnets, nodeSubnets, sourceCIDRs, listenerPort, backendPort, healthCheckPort); err != nil {
```

Comment:
Might be nice to have a concrete structure here as passing around all these individual values is slightly hard to reason about. Perhaps a SecurityListDesiredState struct or similar?

Reply:
#152